### PR TITLE
nextcloud-fixer: update all apps

### PR DESCRIPTION
**Pull request by kyrofa**
Nextcloud upgrades don't update apps properly (see nextcloud/server#18673). This results in a very poor upgrade experience if a bundled app needs to update. This PR works around this issue by updating apps automatically.

### CODE REVIEW

File: src/nextcloud/bin/nextcloud-fixer [hunk 1]
```diff
@@ -24,6 +24,12 @@ if nextcloud_is_installed; then
 	trap 'disable_maintenance_mode' EXIT
 
 	occ -n db:convert-filecache-bigint
+
+	echo "Updating all apps..."
+	if occ -n app:update --all; then
+		# No idea why, but Nextcloud wants an upgrade right after
+		occ -n upgrade
```
Comment (kyrofa) [inline src/nextcloud/bin/nextcloud-fixer]: @kesselb any idea why this is required?

**Comment by kyrofa** [comment]
You can pretty easily test this by installing nextcloud from stable:

    $ sudo snap install nextcloud

Noting that the rightclick app is out of date, and then refreshing to the snap from this PR:

    $ sudo snap refresh nextcloud --channel=beta/pr-1207

Once the refresh completes, you should notice that rightclick is now magically up to date.

**Comment by kyrofa** [comment]
@pachulo I assume you've already updated the rightclick app, so I didn't actually bother building this for arm. Just sanity check the code, please.

File: src/nextcloud/bin/nextcloud-fixer (hunk 1 above)
Comment (kesselb) [reply src/nextcloud/bin/nextcloud-fixer]: Yes. occ app:update download the app updates and extract the code to the right place. Next step is to run occ upgrade routine (migrations, bump app version). I don't know why. Probably a "design" design in the past. The same process for updates via the official updater. Updater will fetch the latest release, put the code in place, run occ upgrade routine.

Comment (kesselb) [reply src/nextcloud/bin/nextcloud-fixer]: After running snap refresh and occ upgrade all apps should be up to date (occ upgrade will check the appstore for updates). But https://github.com/nextcloud/server/blob/master/core/Command/App/Update.php will also return 0 if nothing happens.

Comment (kyrofa) [reply src/nextcloud/bin/nextcloud-fixer]: Ah this is by design then. Seems a little odd, but that makes me feel better! It's okay if the update step returns zero, the upgrade step also returns zero if it's already up to date (despite a comment I have to the contrary, so I think that has been different in the past).

Comment (kyrofa) [reply src/nextcloud/bin/nextcloud-fixer]: I've updated the comments here to reflect reality as I understand it, thank you for the explanation.

**Pull request merged**